src/main/java/gov/nasa/pds/harvest/dao/MetadataWriter.java

```diff
@@ -111,9 +113,34 @@ private void writeBatch() throws Exception
             }
         }
         
-        totalRecords += loader.loadBatch(data);
+        // Load batch
+        Set<String> failedIds = new TreeSet<>();
+        totalRecords += loader.loadBatch(data, failedIds);
         log.info("Wrote " + totalRecords + " product(s)");
         
+        // Update failed counter
+        counter.failedFileCount += failedIds.size();
+
+        // Update product counters
+        for(RegistryDocBatch.NJsonItem item: docBatch.getItems())
+        {
+            if(nonRegisteredIds == null)
+            {
+                if(!failedIds.contains(item.lidvid)) 
+                {
+                    counter.prodCounters.inc(item.prodClass);
+                }
+            }
+            else
```

Comment:
here we could also have "else if" on the same line.

I am raising that because that would make the overarching for loop more readable, but that is not a big deal definitelly.